src/django_otp_webauthn/views.py

```diff
@@ -195,8 +196,10 @@ def complete_auth(self, device: AbstractWebAuthnCredential) -> AbstractUser:
 
         You may override this method to implement custom logic.
         """
-        user = device.user
-        if not self.request.user.is_authenticated:
+        if self.request.user.is_authenticated:
+            user = device.user
+        else:
+            user = auth_authenticate(self.request, web_authn_credential=device)
```

Comment:
**Thought:** wondering what would happen if `request.user` is different from `device.user`. I suppose this could happen when an already logged in user uses a Passkey they have on their device for another user account (imagine someone having access to two different user accounts)

I suppose in this scenario we should also call `auth_authenticate` to authenticate as the new user. Otherwise we'll try to mark 2FA as verified with a device that does not belong to the currently authenticated user.

It's a bit of an edge case and also I might be overthinking it. It's getting late here and my mind is not very fresh.

Let me know what you think.

Reply:
That's a good question. I kept this logically the same as it originally was, which was:
```
user = device.user
if not self.request.user.is_authenticated:
  auth_login(self.request, user)

otp_login(self.request, device)
```

So `device.user` was used already even if it didn't match `self.request.user`.  This is definitely a thing which could happen, though, and we probably should fail it if they are different users. I'm going to give that a test in case it's already handled separately and I just missed it.

Reply:
The more I look at this, the more I think you're right that an error should be returned here. Let's follow up on that separately. I'm not sure what the correct error would be here. An HTTP 400 because it's bad input, the same as you would a bad password if exchanging a username and password for a token via an api? Or a 401 because it's also kind of that and authenticating differently could fix it.